src/main/java/io/vertx/ext/mongo/BulkOperation.java

```diff
@@ -57,27 +51,12 @@ private BulkOperation(BulkOperationType type) {
    */
   public BulkOperation(JsonObject json) {
     String typeValue = json.getString("type");
-    if (typeValue != null)
-      this.type = BulkOperationType.valueOf(typeValue.toUpperCase());
+    type = typeValue != null ? BulkOperationType.valueOf(typeValue.toUpperCase()): null;
```

Comment:
This is another example of change that does not add value but makes the PR grow bigger.

Reply:
imo it improves readability, as below is the same form of null checking implemented for `collation`.
If you prefer the "2-line" form, I'll change both accordingly.
just let me know

Reply:
I understand but if something does not bring obvious value to the user, let's not do it.